contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthValidator.java

```diff
@@ -0,0 +1,473 @@
+{
+  /**
+   * List of accessors (setters) in order of fields in input JSON
+   */
+  List<com.datatorrent.contrib.parser.FixedWidthParser.TypeInfo> setters;
+  /**
+   * Schema is read into this object to access fields
+   */
+  private FixedWidthSchema schema;
+  /**
+   * Input Class for which an object is to be emitted if all validations pass
+   */
+  private Class<?> pojoClass;
+
+  /**
+   * Parameterized constructor to initialize data members
+   * @param schema
+   * @param pojoClass
+   * @param setters
+   */
+  public FixedWidthValidator(FixedWidthSchema schema, Class<?> pojoClass, List<com.datatorrent.contrib.parser.FixedWidthParser.TypeInfo> setters)
+  {
+    this.schema = schema;
+    this.pojoClass = pojoClass;
+    this.setters = setters;
+  }
+
+  /**
+   * Get the Schema
+   * @return
+   */
+  public FixedWidthSchema getSchema()
+  {
+    return schema;
+  }
+
+  /**
+   * Set the Schema
+   * @param schema
+   */
+  public void setSchema(FixedWidthSchema schema)
+  {
+    this.schema = schema;
+  }
+
+  /**
+   * Get the setters
+   * @return
+   */
+  public List<com.datatorrent.contrib.parser.FixedWidthParser.TypeInfo> getSetters()
+  {
+    return setters;
+  }
+
+  /**
+   * Set the Setters
+   * @param setters
+   */
+  public void setSetters(List<FixedWidthParser.TypeInfo> setters)
+  {
+    this.setters = setters;
+  }
+
+  /**
+   * Get the pojoClass
+   * @return
+   */
+  public Class<?> getPojoClass()
+  {
+    return pojoClass;
+  }
+
+  /**
+   * Set the pojoClass
+   * @param pojoClass
+   */
+  public void setPojoClass(Class<?> pojoClass)
+  {
+    this.pojoClass = pojoClass;
+  }
+
+  /**
+   * Function to validate Integer type data and set it in the Object
+   * using accessors in TypeInfo
+   * @param value
+   * @param fieldName
+   * @param typeInfo
+   * @param pojoObject
+   * @param toEmit
+   */
+  public void parseandsetIntegerValue(String value, String fieldName, FixedWidthParser.TypeInfo typeInfo, Object pojoObject, HashMap toEmit)
+  {
+    try {
+      if (value != null && !value.isEmpty()) {
+        int result = Integer.parseInt(value);
+        toEmit.put(fieldName,result);
+        if (typeInfo != null && pojoObject != null) {
+          typeInfo.setter.set(pojoObject, result);
+        }
+      }
+      else
+        toEmit.put(fieldName,value);
+
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("Error parsing" + value + " to Integer type", e);
+    } catch (Exception e) {
+      throw new RuntimeException("Error setting " + value + " in the given class" + typeInfo.toString(), e);
+
+    }
+  }
+
+  /**
+   * Function to validate Double type data and set it in the Object
+   * using accessors in TypeInfo
+   * @param value
+   * @param fieldName
+   * @param typeInfo
+   * @param pojoObject
+   * @param toEmit
+   */
+  public void parseandsetDoubleValue(String value, String fieldName, FixedWidthParser.TypeInfo typeInfo, Object pojoObject, HashMap toEmit)
```

Comment:
This is a validator, its function should not be to set... I would suggest to put this code in operator itself. I don't see much advantage is moving this code out to another class.